src/utils/utils.cpp

```diff
@@ -38,35 +38,48 @@ namespace backtrace_helpers {
     }
 }
 
+void WaitForCompleteFlush() {
+    using namespace std::chrono_literals;
+
+    std::this_thread::sleep_for(500us);
+    Paper::Logger::WaitForFlush();
+    while (Paper::Internal::logQueue.size_approx() > 0) {
+        std::this_thread::sleep_for(500us);
+    }
+}
+
 void safeAbort(const char* func, const char* file, int line, uint16_t frameCount) {
-    static auto logger = Logger::get().WithContext("CRASH_UNLESS");
+    static auto logger = il2cpp_utils::getLogger();
     // we REALLY want this to appear at least once in the log (for fastest fixing)
     for (int i = 0; i < 2; i++) {
         usleep(100000L);  // 0.1s
         // TODO: Make this eventually have a passed in context
-        logger.critical("Aborting in %s at %s:%i", func, file, line);
+        logger.fmtLog<Paper::LogLevel::CRIT>("Aborting in {} at {}:{}", func, file, line);
     }
     logger.Backtrace(frameCount);
-    Logger::closeAll();
     usleep(100000L);  // 0.1s
+    WaitForCompleteFlush();
     std::terminate();  // cleans things up and then calls abort
 }
 
-__attribute__((format(printf, 4, 5))) void safeAbortMsg(const char* func, const char* file, int line, const char* fmt, ...) {
-    static auto logger = Logger::get().WithContext("CRASH_UNLESS");
+void safeAbortMsg(const char* func, const char* file, int line, const char* fmt, ...) {
+    static auto logger = il2cpp_utils::getLogger();
```

Comment:
drop static and avoid copy but also does not matter as much since we are exiting